src/azure-cli-core/azure/cli/core/command_recommender.py

```diff
@@ -292,119 +383,131 @@ def _normalize_parameters(self, raw_parameters):
-            elif 'resource_group' in error_msg or 'resource group' in error_msg:
-                error_type = AladdinUserFaultType.ResourceGroupNotFound
-        elif 'pattern' in error_msg or 'is not a valid value' in error_msg or 'invalid' in error_msg:
-            error_type = AladdinUserFaultType.InvalidParameterValue
-            if 'jmespath_type' in error_msg:
-                error_type = AladdinUserFaultType.InvalidJMESPathQuery
-            elif 'datetime_type' in error_msg:
-                error_type = AladdinUserFaultType.InvalidDateTimeArgumentValue
-            elif '--output' in error_msg:
-                error_type = AladdinUserFaultType.InvalidOutputType
-            elif 'resource_group' in error_msg:
-                error_type = AladdinUserFaultType.InvalidResourceGroupName
-            elif 'storage_account' in error_msg:
-                error_type = AladdinUserFaultType.InvalidAccountName
-        elif "validation error" in error_msg:
-            error_type = AladdinUserFaultType.ValidationError
-
-        return error_type.value
-
-    def _extract_parameter_names(self, parameters):  # pylint: disable=no-self-use
-        """Extract parameter names from the raw parameters.
-        An example: ['-g', 'RG', '-n', 'NAME'] ==> ['-g', '-n']
-        """
-
-        from azure.cli.core.commands import AzCliCommandInvoker
-
-        return AzCliCommandInvoker._extract_parameter_names(parameters)  # pylint: disable=protected-access
-
-    def _replace_parameter_values(self, command):
-        """Replace the parameter values in recommended command with values in user's command
-        An example:
-            recommended command: 'az vm create -n MyVm -g MyResourceGroup --image CentOS'
-            user's command:  'az vm create --name user_vm -g user_rg'
-            ==> 'az vm create -n user_vm -g user_rg --image CentOS'
+    def _get_param_mappings(self):
+        try:
+            cmd_table = self.cli_ctx.invocation.commands_loader.command_table.get(self.command, None)
+        except AttributeError:
+            cmd_table = None
+
+        return get_parameter_mappings(cmd_table)
+
+
+def get_parameter_mappings(command_table):
+    """Get the short option to long option mappings of a command
+
+    :param parameter_table: CLI command object
+    :type parameter_table: knack.commands.CLICommand
+    :param command_name: The command name
+    :type command name: str
+    :return: The short to long option mappings of the parameters
+    :type: dict
+    """
+
+    from knack.deprecation import Deprecated
+
+    parameter_table = None
+    if hasattr(command_table, 'arguments'):
+        parameter_table = command_table.arguments
+
+    param_mappings = {
+        '-h': '--help',
+        '-o': '--output',
+        '--only-show-errors': None,
+        '--help': None,
+        '--output': None,
+        '--query': None,
+        '--debug': None,
+        '--verbose': None
+    }
```

Comment:
What kind of parameters need to be put here?
How about `--yes`, `--no-wait`?

Reply:
It's a good catch, let me confirm. I think we need to add all the global parameters here.